docs/core/dependency-loading/understanding-assemblyloadcontext.md

```diff
@@ -0,0 +1,81 @@
+
+It solves the most common dependency loading scenarios.
+
+## How does <xref:System.Runtime.Loader.AssemblyLoadContext> support dynamic dependencies?
+
+<xref:System.Runtime.Loader.AssemblyLoadContext> has various events and virtual functions that can be overridden.
+
+The <xref:System.Runtime.Loader.AssemblyLoadContext.Default?displayProperty=nameWithType> instance only supports overriding the events.
+
+The articles
+[Managed assembly loading algorithm](loading-managed.md),
+[Satellite assembly loading algorithm](loading-resources.md), and
+[Unmanaged (native) library loading algorithm](loading-unmanaged.md) refer to all the available events and virtual functions.  The articles show each event and functions relative position in the loading algorithms. This article won't reproduce that information.
+
+This section will cover the general principles for the relevant events and functions.
+
+- **Be repeatable**. It's required that a query for a specific dependency will always result in the same response. The same loaded dependency instance must be returned. This requirement is fundamental  for cache consistency.
+  - The exception is that the initial `null` response could be followed by a later non-`null` response.
+  - For managed assemblies in particular, we're creating a <xref:System.Reflection.AssemblyName.Name?displayProperty=nameWithType> to <xref:System.Reflection.Assembly> cache. There can be exactly one <xref:System.Reflection.AssemblyName.Name?displayProperty=nameWithType> in the cache.
+- **Typically don't throw**.  It's expected that these functions return `null` rather than throw when unable to find the requested dependency. Throwing should be restricted to unexpected errors like a corrupt assembly or out of memory.
+- **Avoid recursion**. Beware that these functions and handlers are implementing the loading rules for locating dependencies. Your implementation shouldn't call APIs that trigger recursion. Your code should typically be calling **AssemblyLoadContext** load functions that require a specific path or memory reference argument.
+- **Load into the correct AssemblyLoadContext**. The choice of where to load dependencies is application-specific.  The choice is implemented by these events and functions. When your code calls **AssemblyLoadContext** load-by-path functions call them on the instance where you want the code loaded. Sometime returning `null` and letting the <xref:System.Runtime.Loader.AssemblyLoadContext.Default?displayProperty=nameWithType> handle the load may be the simplest option.
+
+## How are dynamic dependencies isolated?
+
+One of the primary purposes of exposing the <xref:System.Runtime.Loader.AssemblyLoadContext> is to allow for isolation of dependencies.
+
+The general static dependency model requires a one to one mapping of name to dependency instance.
```

Comment:
I find this whole section very confusing. I think I can guess what you're trying to say - but I know most of what this describes. Somebody new to this area will probably be lost. This uses lot of new "Terms" like "dependency instance" (what is it), "static/dynamic dependencies" (not explained anywhere), "binary isolation", ...

I'll think about this some more, but in general I would completely rewrite this part in a more "readable" way, even is it's going to be lot longer.